extensions/interactions/InteractiveMap/directives/oppia-interactive-interactive-map.component.ts

```diff
@@ -0,0 +1,195 @@
+      this.playerPositionService.onNewCardAvailable.subscribe(
+        () => {
+          this.interactionIsActive = false;
+          this.setOverlay();
+        }
+      )
+    );
+
+    const { latitude, longitude, zoom } = (
+      this.interactionAttributesExtractorService.getValuesFromAttributes(
+        'InteractiveMap',
+        this._getArgs()
+      ) as InteractiveMapCustomizationArgs
+    );
+    this.coords = [latitude.value, longitude.value];
+    if (latitude.value === undefined || longitude.value === undefined) {
+      this.coords = [0, 0];
+    }
+    this.zoom = zoom.value;
+    this.zoomLevel = parseInt(this.zoom + '', 10) || 0;
+    this.interactionIsActive = (this.lastAnswer === null);
+    this.mapOptions = {
+      layers: [tileLayer(
+        this.optionsSpec.layers[0].url,
+        { attribution: this.optionsSpec.layers[0].attribution }
+      )],
+      zoom: this.zoomLevel,
+      center: latLng([this.coords[0], this.coords[1]])
+    };
+    if (!this.interactionIsActive) {
+      this.changeMarkerPosition(
+        this.lastAnswer[0], this.lastAnswer[1]);
+    }
+  }
+
+  private _getArgs() {
+    return {
+      latitudeWithValue: this.latitudeWithValue,
+      longitudeWithValue: this.longitudeWithValue,
+      zoomWithValue: this.zoomWithValue
+    };
+  }
+
+  private changeMarkerPosition(lat, lng) {
+    const newMarker = marker(
+      [lat, lng],
+      {
+        icon: icon({
+          iconUrl: this.urlInterpolationService.getExtensionResourceUrl(
+            '/interactions/InteractiveMap/static/marker-icon.png'),
+          // The size of the icon image in pixels.
+          iconSize: [25, 41],
+          // The coordinates of the "tip" of the icon.
+          iconAnchor: [12, 41],
+          shadowUrl: this.urlInterpolationService.getExtensionResourceUrl(
+            '/interactions/InteractiveMap/static/marker-shadow.png'),
+          // The size of the shadow image in pixels.
+          shadowSize: [41, 41],
+          // The coordinates of the "tip" of the shadow.
+          shadowAnchor: [13, 41],
+          // The URL to a retina sized version of the icon image.
+          // Used for Retina screen devices.
+          iconRetinaUrl: this.urlInterpolationService.getExtensionResourceUrl(
+            '/interactions/InteractiveMap/static/marker-icon-2x.png'),
+          shadowRetinaUrl: (
+            this.urlInterpolationService.getExtensionResourceUrl(
+              '/interactions/InteractiveMap/static/marker-shadow.png'))
+        })
+      }
+    );
+    this.mapMarkers = newMarker;
+  }
+
+  leafletClick(e: LeafletMouseEvent): void {
+    if (this.interactionIsActive) {
+      const newLat = e.latlng.lat;
+      const newLng = e.latlng.lng;
+      this.changeMarkerPosition(newLat, newLng);
+      this.currentInteractionService.onSubmit(
+        [newLat, newLng] as unknown as string,
+        this.interactiveMapRulesService as unknown as InteractionRulesService
+      );
+    }
```

Comment:
```suggestion
    if (!this.interactionIsActive) {
      return
    }
    
    const newLat = e.latlng.lat;
    const newLng = e.latlng.lng;
    this.changeMarkerPosition(newLat, newLng);
    this.currentInteractionService.onSubmit(
      [newLat, newLng] as unknown as string,
      this.interactiveMapRulesService as unknown as InteractionRulesService);
```
[Optional] This will help to have less right indented code.

Reply:
I will probably do it sometime later, I have noticed that during migration, code refactoring has caused some regressions. So currently not doing such things with migration-related PRs to avoid reverts.